src/rdbms/rdbms_queries.erl

```diff
@@ -89,35 +89,37 @@ execute_upsert(Host, Name, InsertParams, UpdateParams, UniqueKeyValues) ->
 
 %% @doc
 %% This functions prepares query for inserting a row or updating it if already exists
+%% Updates can be either fields or literal expressions like "column = tab.column + 1"
 -spec prepare_upsert(Host :: mongoose_rdbms:server(),
                      QueryName :: atom(),
                      TableName :: atom(),
                      InsertFields :: [binary()],
-                     UpdateFields :: [binary()],
```

Comment:
could be an extra argument `UpdateSQL`, which is empty by default.
And so we would have prepare_upsert/6 and prepare_upsert/7.

But  not thaaat critical, it's called once per statement anyway.

Reply:
I admit I avoided adding the extra argument. I could do it like you suggest to make it more obvious for the user.